packages/react-log-viewer/src/LogViewer/loggerToolbar.tsx

```diff
@@ -0,0 +1,177 @@
+  dataSourcesAmount?: number;
+  dataSourceTitles?: string[] | null[];
+  customToolbarActions?: () => React.ReactNode | React.ReactNode[];
+  scrollToRow: (searchedRow: number) => void;
+}
+
+const LoggerToolbar: React.FC<LoggerToolbarProps> = ({
+  scrollToRow,
+  // customToolbarActions,
+  dataSourcesAmount,
+  includesPlay,
+  includesDownload,
+  includesFullScreen,
+  includesLaunchExternal
+}) => {
+  const [currentSearchItemCount, setCurrentSearchItemCount] = useState<number>(1); // need to have a function not to display these unless there are any found searches
+  const loggerState = useLoggerContext();
+  const {
+    searchedWordIndexes,
+    setSearchedWordIndexes,
+    rowInFocus,
+    setRowInFocus,
+    currentDataSource,
+    setCurrentDataSource,
+    searchedInput,
+    setSearchedInput,
+    dataSourceTitles,
+    setDataSourceTitles,
+    foundWordIndex,
+    setFoundWordIndex
+  } = loggerState;
+  const DEFAULT_FOCUS = -1;
+  const DEFAULT_INDEX = 0;
+
+  /* Defaulting the first focused row that contain searched keywords */
+  useEffect(() => {
+    if (searchedWordIndexes.length >= 1) {
+      setFoundWordIndex(DEFAULT_INDEX);
+    }
+  }, [searchedWordIndexes]);
+
+  /* Defaulting the names of the dropdown items if the default toolbar */
+  useEffect(() => {
+    if (dataSourcesAmount > 1 && dataSourceTitles[currentDataSource] === 'Default') {
+      let titleCounter: number = 0;
+      const newTitles: string[] = [];
+      while (titleCounter < dataSourcesAmount) {
+        newTitles.push(`Data Source ${titleCounter + NUMBER_INDEX_DELTA}`);
+        ++titleCounter;
+      }
+
+      setDataSourceTitles(newTitles);
+    }
+  }, [dataSourcesAmount]);
+
+  /* Clearing out the search input */
+  const handleClear = (): void => {
+    setSearchedInput('');
+    setFoundWordIndex(0);
+    setSearchedWordIndexes([]);
+    setRowInFocus(DEFAULT_FOCUS);
+  };
+
+  /* Moving focus over to next row containing searched word */
+  const handleNextSearchItem = (): void => {
+    const oldIndex = searchedWordIndexes.indexOf(rowInFocus);
+    const adjustedSearchCount = currentSearchItemCount + 1;
+    let temp = foundWordIndex;
+
+    if (oldIndex >= searchedWordIndexes.length - NUMBER_INDEX_DELTA) {
+      return null;
+    }
+
+    setCurrentSearchItemCount(adjustedSearchCount);
+    setFoundWordIndex(++temp);
+    scrollToRow(searchedWordIndexes[foundWordIndex]);
+  };
+
+  /* Moving focus over to next row containing searched word */
+  const handlePrevSearchItem = (): void => {
+    const oldIndex = searchedWordIndexes.indexOf(rowInFocus);
+    const adjustedSearchCount = currentSearchItemCount - 1;
+    let temp = foundWordIndex;
+
+    if (oldIndex < 0) {
+      return null;
+    }
+
+    setCurrentSearchItemCount(adjustedSearchCount);
+    setFoundWordIndex(--temp);
+    scrollToRow(searchedWordIndexes[--temp]);
+  };
+
+  /* Way to make this logic much simpler? */
+  const toolbarItems = (): React.ReactNode => (
```

Comment:
Make this an expression like `const toolbarItems = <>{stuff}</>` instead of a function like `const toolbarItems = () => {}`.

It's more performant for state and simpler.